consensus/tendermint/tendermint.go

```diff
@@ -0,0 +1,436 @@
+package tendermint
+
+import (
+	"maps"
+	"slices"
+	"sync"
+	"time"
+
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+type step uint8
+
+const (
+	propose step = iota
+	prevote
+	precommit
+)
+
+func (s step) String() string {
+	switch s {
+	case propose:
+		return "propose"
+	case prevote:
+		return "prevote"
+	case precommit:
+		return "precommit"
+	default:
+		return "unknown"
+	}
+}
+
+const (
+	maxFutureHeight = uint(5)
+	maxFutureRound  = uint(5)
```

Comment:
I think these should be configurable, probably inside the `Tendermint` struct

Reply:
I don't think we should as this can lead to an attack vector, so I didn't want to make it configurable for the time being.

Reply:
By configurable, I didn't mean the node user can configure it via command flag. What I'm understanding here is that the limit of future height and future round is not hardcoded in the tendermint algorithm, but by our application level. Then even if we have this and disallow the node to customize this, it should be on other level.

Reply:
I don't think it is worth doing it right now, since I am not sure how it will be used.